pym/bob/generators/VisualStudio.py

```diff
@@ -278,7 +278,7 @@ def generate(self, extra, bobRoot):
 
         # gather root paths
         bobPwd = Path(os.getcwd())
-        if sys.platform == 'msys':
+        if isMsys:
```

Comment:
You must actually call the function:

```suggestion
        if isMsys():
```

Also see the other changes with the same problem. You also forgot to import it so this will certainly fail at runtime.

Reply:
ohh?! i tested it successfully, so this was just luck? :D

Reply:
A function is always True ;):

```
>>> def foo():
...   pass
... 
>>> bool(foo)
True
```